org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/internal/EventHub.java

```diff
@@ -60,15 +60,27 @@ public void start(VirtualMachine vm) {
                     }
 
                     EventSet set = queue.remove();
+
+                    // Print JDI Events
+                    StringBuffer buf = new StringBuffer("\nJDI Event Set: {");
+                    for (Event event : set) {
+                        buf.append(event);
+                        buf.append(", ");
+                    }
+                    buf.append("}\n");
+                    Logger.logInfo(buf.toString());
+
                     boolean shouldResume = true;
+                    boolean beConsumed = false;
```

Comment:
In current EventHub implementation, if an event is not consumed by any handler, it'll always call eventSet.resume() to resume VM due to the default true of shouldResume. Add another flag beConsumed to let EventHub only to make resume decision for those handled event. And ignore those not handled event. 
Of course, we can discuss if this flag is necessary or not.